detection_rules/integrations.py

```diff
@@ -89,6 +159,49 @@ def find_least_compatible_version(package: str, integration: str,
     raise ValueError(f"no compatible version for integration {package}:{integration}")
 
 
+def find_compatible_version_window(package: str, integration: str,
```

Comment:
We should expose this to a command to print out the matrix to have a snapshot at any given time